src/gui/maptools/qgsmaptooladvanceddigitizing.cpp

```diff
@@ -82,6 +90,10 @@ void QgsMapToolAdvancedDigitizing::canvasReleaseEvent( QgsMapMouseEvent *e )
 
 void QgsMapToolAdvancedDigitizing::canvasMoveEvent( QgsMapMouseEvent *e )
 {
+  if ( canvas()->currentLayer() && !canvas()->currentLayer()->isSpatial() )
+  {
+    return;
+  }
```

Comment:
Why are these blocks needed? You're already disabling the cad dock widget in the onCurrentLayerChanged() function.

Reply:
@nirvn, my bad. Yes you're right- they're not needed. In my older PR, I wasn't disabling the cad dock widget (only clearing it). I added the `disable()` call here which I think is the right approach, but forgot to remove those checks.
I have just finished my day job so I've now removed them & updated the branch.
Thanks for looking at this- it's much appreciated.